src/aof.c

```diff
@@ -1160,16 +1175,24 @@ int loadSingleAppendOnlyFile(char *filename) {
 
         /* Serve the clients from time to time */
         if (!(loops++ % 1000)) {
-            loadingProgress(ftello(fp));
+            progress_size = ftello(fp) - base_size;
+            loadingIncrAofProgress(progress_size);
+            base_size += progress_size;
             processEventsWhileBlocked();
             processModuleLoadingProgressEvent(1);
         }
 
         if (fgets(buf,sizeof(buf),fp) == NULL) {
-            if (feof(fp))
+            if (feof(fp)) {
+                /* When we read the end of the file, we also need to update 
+                 * the progress_size. Or the `server.loading_loaded_bytes` 
+                 * statistics may be inaccurate. */
+                progress_size = ftello(fp) - base_size;
+                loadingIncrAofProgress(progress_size);
```

Comment:
maybe this better be after the loop rather than inside the condition that breaks?

Reply:
OK